opflexagent/gbp_ovs_agent.py

```diff
@@ -10,365 +10,187 @@
-    cfg.BoolOpt('hybrid_mode',
-                default=False,
-                help=_("Whether Neutron's ports can coexist with GBP owned"
-                       "ports.")),
-    cfg.StrOpt('epg_mapping_dir',
-               default='/var/lib/opflex-agent-ovs/endpoints/',
-               help=_("Directory where the EPG port mappings will be "
-                      "stored.")),
-    cfg.StrOpt('as_mapping_dir',
-               default='/var/lib/opflex-agent-ovs/services/',
-               help=_("Directory where the anycast svc mappings will be "
-                      "stored.")),
-    cfg.StrOpt('opflex_agent_dir',
-               default='/var/lib/neutron/opflex_agent',
-               help=_("Directory where the opflex agent state will be "
-                      "stored.")),
-    cfg.ListOpt('opflex_networks',
-                default=['*'],
-                help=_("List of the physical networks managed by this agent. "
-                       "Use * for binding any opflex network to this agent")),
-    cfg.ListOpt('internal_floating_ip_pool',
-               default=['169.254.0.0/16'],
-               help=_("IP pool used for intermediate floating-IPs with SNAT")),
-    cfg.ListOpt('internal_floating_ip6_pool',
-               default=['fe80::/64'],
-               help=_("IPv6 pool used for intermediate floating-IPs "
-                      "with SNAT"))
-]
-cfg.CONF.register_opts(gbp_opts, "OPFLEX")
-
-FILE_EXTENSION = "ep"
-FILE_NAME_FORMAT = "%s." + FILE_EXTENSION
-VRF_FILE_EXTENSION = "rdconfig"
-VRF_FILE_NAME_FORMAT = "%s." + VRF_FILE_EXTENSION
-METADATA_DEFAULT_IP = '169.254.169.254'
-METADATA_SUBNET = '169.254.0.0/16'
-
 
 class GBPOvsPluginApi(rpc.GBPServerRpcApiMixin):
     pass
 
 
-class ExtSegNextHopInfo(object):
-    def __init__(self, es_name):
-        self.es_name = es_name
-        self.ip_start = None
-        self.ip_end = None
-        self.ip_gateway = None
-        self.ip6_start = None
-        self.ip6_end = None
-        self.ip6_gateway = None
-        self.next_hop_iface = None
-        self.next_hop_mac = None
-        self.from_config = False
-        self.uuid = uuidutils.generate_uuid()
-
-    def __str__(self):
-        return ("%s: ipv4 (%s-%s,%s), ipv6 (%s-%s,%s), if %s, mac %s, (%s)" %
-            (self.es_name, self.ip_start, self.ip_end, self.ip_gateway,
-            self.ip6_start, self.ip6_end, self.ip6_gateway,
-            self.next_hop_iface, self.next_hop_mac,
-            "configured" if self.from_config else "auto-allocated"))
+class GBPOpflexAgent(sg_rpc.SecurityGroupAgentRpcCallbackMixin,
+                     rpc.GbpNeutronAgentRpcCallbackMixin):
+    """Group Based Policy Opflex Agent.
 
-    def is_valid(self):
-        return ((self.ip_start and self.ip_gateway) or
-                (self.ip6_start and self.ip6_gateway))
+    The GBP opflex agent assumes a pre-existing bridge (integration bridge is
+    not required). This agent is an interface between the SDN controller and
```

Comment:
The second sentence reads wrong. How about:

This agent is an interface between Openstack Neutron and agent-ovs, which renders policies from an OpFlex-based SDN controller (like Cisco ACI) into OpenFlow rules for OVS.